### ISSUE TITLE
Using local data for heatmap

### ISSUE THREAD

**Issue by jlguidubaldi1** [gosling-lang/gos#121]
I'm having trouble using gos to generate a heatmap of .cool data I have on my local device. I've tried specifying a local path from my machine's hard drive, a local path from my platform (ie uploading my data to Google Colab then running the code from there), as well as uploading my data to Google Drive, mounting Drive to Colab, and trying to specify a path from the mounted Drive. All of these result in the successful generation of an empty heatmap plot, but no data is visible.

That said, I am able to successfully generate a complete interactive heatmap using the test code and data supplied.

I'm wondering if maybe it's because I'm not specifying the data as a URL, as this is the only difference between the test code and my own code? Anyone have any thoughts?

**Comment by manzt** [comment]
Hi there, thanks for trying out gos. Without more information it's very challenging to try to debug your particular issue. 

First, cooler files must be multi-resolution (i.e., `.mcool`). Have you run [`cooler zoomify`](https://docs.higlass.io/data_preparation.html#zoomify)?

**Comment by jlguidubaldi1** [comment]
I have tried inputting my data both in a .cool and in a .mcool format, and received the same result from both.

**Comment by manzt** [comment]
Hi, just wanted to send a response that I haven't forgotten about this! It isn't totally clear what the issue seems to be, and it might take some time to for me to diagnose and fix. I have been able to reproduce the error for coolers, so it is likely not an issue on your end but potentially a regression in our Colab support.

**Comment by jlguidubaldi1** [comment]
Just wanted to check in on this again. Any chance progress has been made?

**Comment by manzt** [comment]
Hi there. The most recent release of Gos includes some refactoring of the server which should fix this issue (see #126 ). I just opened Google Colab and viewed a "local" cooler.

```python
!pip install gosling[all]
!wget https://osf.io/3h9js/download -qO ./test.mcool

import gosling as gos

size = 500
data = gos.matrix("./test.mcool")

track = gos.Track(data).mark_bar().encode(
  x=gos.X("xs:G", axis="bottom"),
  xe="xe:G",
  y=gos.Y("ys:G", axis="left"),
  ye="ye:G",
  color=gos.Color("value:Q", range="hot", legend=True),
).properties(width=size, height=size)

track.view()
```

**Comment by manzt** [comment]
Closing from #126. Hopefully this fixed the issue, otherwise please feel free to re-open. Apologies this took some time to get to!
